lib/ansible/modules/clustering/kubevirt/k8s_service.py

```diff
@@ -0,0 +1,208 @@
+        default: present
+        choices:
+            - present
+            - absent
+        type: str
+    name:
+        description:
+            - Use to specify a Service object name.
+        required: true
+        type: str
+    namespace:
+        description:
+            - Use to specify a Service object namespace.
+        required: true
+        type: str
+    selector:
+        description:
+            - TODO
+        default: []
+        type: dict
+    type:
+        description:
+            - TODO
+        default: ClusterIP
+        choices:
+            - NodePort
+            - ClusterIP
+            - LoadBalancer
+    ports:
+        description:
+            - TODO
+        required: true
+        type: list
+    labels:
+        description:
+            - Labels are key/value pairs that are attached to objects.
+        default: {}
+        type: dict
+    spec:
+        description:
+            - Use to specify any additional parameters to be passed to the cluster. Everything defined here takes precedence over this module's other parameters.
+        default: {}
+        type: dict
+
+extends_documentation_fragment:
+  - k8s_auth_options
+  - k8s_resource_options
+
+requirements:
+  - python >= 2.7
+  - openshift >= 0.6.2
+'''
+
+EXAMPLES = '''
+TODO
+'''
+
+RETURN = '''
+service:
+    description:
+      - The Service object.
+    returned: success
+    type: complex
+'''
+
+import copy
+import traceback
+
+from collections import defaultdict
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC, COMMON_ARG_SPEC
+from ansible.module_utils.k8s.raw import KubernetesRawModule
+
+from openshift import watch
+from openshift.dynamic.client import ResourceInstance
+from openshift.helper.exceptions import KubernetesException
+
+
+SERVICE_ARG_SPEC = {
+    'merge_type': {'type': 'list', 'choices': ['json', 'merge', 'strategic-merge']},
+    'state': {
+        'type': 'str',
+        'choices': [
+            'present', 'absent'
+        ],
+        'default': 'present'
+    },
+    'selector': {'type': 'dict', 'default': {}},
+    'type': {
+        'type': 'str',
+        'choices': [
+            'NodePort', 'ClusterIP', 'LoadBalancer'
+        ],
+        'default': 'ClusterIP'
+    },
+    'ports': {'type': 'list', 'default': []},
+    'labels': {'type': 'dict', 'default': {}},
+    'spec': {'type': 'dict', 'default': {}},
+}
+
+API_VERSION = 'v1'
+
+
+def virtdict():
+    return defaultdict(virtdict)
+
+
+class KubernetesService(KubernetesRawModule):
+    def __init__(self, *args, **kwargs):
+        super(KubernetesService, self).__init__(*args, **kwargs)
+
+    @staticmethod
+    def merge_dicts(x, y):
```

Comment:
We already have this method [here](https://github.com/kubevirt/ansible-kubevirt-modules/blob/dce8831abbe04174d098ee00c0a0dbd61a701cdf/lib/ansible/modules/clustering/kubevirt/kubevirt_vm.py#L228). Please reuse.

Reply:
If upstream rejects a merge request, then I'll separate this into common code. But with the goal of  submitting this upstream, I can't extract this method (yet).

Reply:
There is no need to copy this code. It seems like good reason to create a utility class.

Reply:
I can't create a `module_utils/kubevirt/utils.py`, have `k8s_service.py` depend on its contents and then send a PR adding `k8s_service.py` to upstream ansible. They'd reject it with _there's code missing here_. _If_ they reject the PR and we end up keeping `k8s_service.py`_ in our own repo, _then_ I can have it use kubevirt's utility code. Not before.

Reply:
I am ok with doing it in separate PR. In general please do not copy any code.